team/bundles/org.eclipse.team.core/src/org/eclipse/team/internal/core/subscribers/AbstractContentComparator.java

```diff
@@ -52,30 +52,16 @@ public boolean compare(IResource e1, IFileRevision e2, IProgressMonitor monitor)
 	}
 
 	private boolean compareObjects(Object e1, Object e2, IProgressMonitor monitor) {
-		InputStream is1 = null;
-		InputStream is2 = null;
-		try {
-			monitor.beginTask(null, 100);
-			is1 = getContents(e1, Policy.subMonitorFor(monitor, 30));
-			is2 = getContents(e2, Policy.subMonitorFor(monitor, 30));
+		monitor.beginTask(null, 100);
+		try (InputStream is1 = getContents(e1, Policy.subMonitorFor(monitor, 30));
+				InputStream is2 = getContents(e2, Policy.subMonitorFor(monitor, 30))) {
 			return contentsEqual(Policy.subMonitorFor(monitor, 40), is1, is2, shouldIgnoreWhitespace());
+		} catch (IOException e) {
+			return false;
```

Comment:
Same as above: This is not behavior preserving? Before, with an `IOException` during `close()` the method still returned whatever `contentEquals(...)` delivered, now it returns `false`.